### PR TITLE
MaxChainedCallsOnSameLine: don't count package references as chained calls

### LINKED ISSUE #5028: MaxChainedCallsOnSameLine reports fully-qualified class names
## Expected Behavior
The `MaxChainedCallsOnSameLine` should probably ignore fully-qualified class names (like this [example](https://github.com/detekt/detekt/blob/ddcf77d9d4651bd117ff622180312141f900ac23/detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/FormattingRule.kt#L92) from detekt's codebase). These call chains are somewhat different in nature from its intended purpose, and the rule probably shouldn't have a second effect of restricting fully-qualified class names, which may be necessary (or at least preferred over [named imports](https://kotlinlang.org/docs/packages.html#imports) with `as`) when classes with the same name but different packages are used. See discussion in #5020.

A fix would likely require type resolution to determine the role of chained calls, but this complicates the rule and makes it harder to be applied, just to avoid a fairly rare edge case. So I wanted to open this issue for discussion/feedback on how to proceed.

## Observed Behavior
`MaxChainedCallsOnSameLine` reports fully qualified class names above its `maxChainedCalls` limit.

## Your Environment
* Version of detekt used: 1.21.0-RC2

### PR DESCRIPTION
**Pull request by t-kameyama**
Fixes #5028

### CODE REVIEW

**Review by amitdash291 — APPROVED** [review]

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxChainedCallsOnSameLineSpec.kt [hunk 1]
```diff
@@ -1,58 +1,61 @@
...
-class MaxChainedCallsOnSameLineSpec {
+@KotlinCoreEnvironmentTest
+class MaxChainedCallsOnSameLineSpec(private val env: KotlinCoreEnvironment) {
     @Test
     fun `does not report 2 calls on a single line with a max of 3`() {
         val rule = MaxChainedCallsOnSameLine(TestConfig(mapOf("maxChainedCalls" to 3)))
         val code = "val a = 0.plus(0)"
 
-        assertThat(rule.compileAndLint(code)).isEmpty()
+        assertThat(rule.lintWithContext(env, code)).isEmpty()
```
Comment (BraisGabin) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxChainedCallsOnSameLineSpec.kt:18]: Is it possible to use `compileLintWithContext` in all this cases?

**Review by BraisGabin — APPROVED** [review]
Thanks!

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxChainedCallsOnSameLineSpec.kt (hunk 1 above)
Comment (t-kameyama) [reply detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MaxChainedCallsOnSameLineSpec.kt]: [0e537f8](https://github.com/detekt/detekt/pull/5036/commits/0e537f8ed1dfd44e640d8667be87bdb64b23113f)

**Comment by TWiStErRob** [comment]
Does this also fix import statements?

https://github.com/TWiStErRob/net.twisterrob.sun/security/code-scanning/275

**Comment by BraisGabin** [comment]
That was fixed on other pr.

**Pull request merged**